server/src/main/java/org/elasticsearch/action/search/SearchType.java

```diff
@@ -50,6 +50,11 @@
      */
     public static final SearchType DEFAULT = QUERY_THEN_FETCH;
 
+    /**
+     * Non-deprecated types
+     */
+    public static final SearchType [] CURRENTLY_SUPPORTED = {QUERY_THEN_FETCH, DFS_QUERY_THEN_FETCH};
```

Comment:
curious about this: what problems were there with deprecated values? Given that it's only used in tests, maybe this could be only in the test where it's needed?

Reply:
The test was randomly picking from enum's values but the "fromString" code in the same enum would barf on one of those values.

>Given that it's only used in tests, maybe this could be only in the test where it's needed?

I thought there would be less chance of getting it wrong in future if the enum where the values are decoded also listed the acceptable values it can decode.